File: shared/src/main/java/ru/privatenull/pnauth/api/AuthTotpApi.java
package ru.privatenull.pnauth.api;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

/** Optional second-factor operations. */
public interface AuthTotpApi {
    CompletableFuture<TotpSetup> beginTotpSetup(UUID uniqueId, String password, String issuer);
    CompletableFuture<AuthResult> confirmTotpSetup(UUID uniqueId, String code);
    CompletableFuture<AuthResult> verifyTotp(UUID uniqueId, String code);
    CompletableFuture<AuthResult> disableTotp(UUID uniqueId, String password, String code);
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthTotpApi.kt
package ru.privatenull.pnauth.api

import java.util.UUID
import java.util.concurrent.CompletableFuture

/** Optional second-factor operations. */
interface AuthTotpApi {
    fun beginTotpSetup(uniqueId: UUID, password: String, issuer: String): CompletableFuture<TotpSetup>
    fun confirmTotpSetup(uniqueId: UUID, code: String): CompletableFuture<AuthResult>
    fun verifyTotp(uniqueId: UUID, code: String): CompletableFuture<AuthResult>
    fun disableTotp(uniqueId: UUID, password: String, code: String): CompletableFuture<AuthResult>
}
